tools/dreport.d/ibm.d/gendumpheader

```diff
@@ -74,10 +74,18 @@ function dump_size () {
 #Function to set dump id to 8 bytes format
 function get_dump_id () {
     # shellcheck disable=SC2154
+    if [ -z $dump_id ]; then
+        dump_id="00000000"
+    fi
     size=${#dump_id}
     if [ "$1" == "$OP_DUMP" ]; then
         msize=$(( size / 2 ))
-        nulltoadd=$(( SIZE_4 - msize ))
+        msize_rem=$(( size % 2 ))
+        if [ "$msize_rem" -ne 0 ]; then
```

Comment:
Hypothetical it will never happen,  What if the dump_id is of 11 or 9 digits ? In that case you will be adding only 1st 8 digits. Better check for size should not be "< 8 > " if that is the case encountered add 0.0 and proceed.

Reply:
But I was able to replicate the problem for the dump IDs which size was an odd number like [0-9], [100-999] etc. Also as it has been made sure that the dump_id is of max 8 characters so no question of getting something which is greater than 8 characters (if it is coming then it could be another critical bug elsewhere). I tested it for every possible scenarios within the limit of 8 characters dump_id and it had worked for them.

Reply:
Please correct me if I am missing something here

Reply:
@aravynd is there isn;t better way to do in bash scripting? 
c++ we do setw(8), setfill(0) to format

Reply:
#!/bin/bash
n=500
printf "%7d\n" ${n} |  tr ' ' '0' 


https://developerfacts.com/answer/3464147-setw-and-setfill-equivalent-in-bash

Reply:
in the commit message remove all these extra unrelated lines, it should not be so detailed

00000040  53 45 43 54 49 4f 4e 20  00 30 00 00 00 00 00 00  |SECTION .0......|
00000050  00 00 00 00 00 02 00 00  00 00 00 00 00 00 04 00  |................|
00000060  44 55 4d 50 20 53 55 4d  4d 41 52 59 00 00 00 00  |DUMP SUMMARY....|
00000070  53 45 43 54 49 4f 4e 20  00 30 00 02 00 00 00 00  |SECTION .0......|
00000080  00 00 00 00 00 02 00 00  00 00 00 00 00 63 92 92  |.............c..|
00000090  48 41 52 44 57 41 52 45  20 44 41 54 41 00 00 00  |HARDWARE DATA...|
000000a0  53 45 43 54 49 4f 4e 20  00 30 00 02 00 00 00 00  |SECTION .0......|
000000b0  00 00 00 01 00 02 00 00  00 00 00 00 00 00 00 00  |................|